builders/interact/ethereum-api/dev-env/foundry.md

```diff
@@ -0,0 +1,375 @@
+
+You will need to create a Foundry project if you don't already have one. You can create one by completing the following steps:
+
+1. Install Foundry with the below commands. The Windows instructions are noticeably different as you'll have to install Rust and then build Foundry from source
+
+    === "Ubuntu"
+
+        ```bash
+        curl -L https://foundry.paradigm.xyz | bash foundryup
+        ```
+
+    === "MacOS"
+
+        ```bash
+        curl -L https://foundry.paradigm.xyz | bash foundryup
+        ```
+
+    === "Windows"
+
+        ```bash
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh
+        cargo install --git https://github.com/foundry-rs/foundry foundry-cli anvil --bins --locked
+        ```
+
+     After installing, you may need to restart your terminal session or add `foundryup` to your PATH. 
+
+2. Create the project, which will create a folder with three folders within it:
+    ```bash
+forge init foundry
+    ```
+
+You may get an error, such as `The target directory is a part of or on its own an already initialized git repository,
+and it requires clean working and staging areas, including no untracked files.` To solve this, you can add files and make a commit if you are maintaining this project within a GitHub repository. Otherwise, you can make a dummy commit without pushing. If you run `forge init foundry` once more, you'll no longer have the error. 
+
+With the default project created, you should see three folders.  
+
+- `lib` - all of the project's dependencies in the form of git submodules
+- `src` - where to put your smart contracts (with functionality)
+- `test` - where to put the forge tests for your project, which are written in Solidity
+
+In addition to these three folders, a git project will also be created along with a prewritten `.gitignore` file with relevant file types and folders ignored.
+
+## The Source Folder {: #the-src-folder }
+
+The preconfigured `foundry` repo includes `Counter.sol` in the `src` folder, as well as a `Counter.s.sol` in the `script` folder and `Counter.t.sol` in the `test` folder. You should delete these files to avoid errors when trying to compile and deploy `MyToken.sol`. You can do so with the following command:
+
+```bash
+rm src/Counter.sol script/Counter.s.sol test/Counter.t.sol
+```
+
+In the following steps, you will be deploying an ERC-20 contract. In the contracts directory, you can create the `MyToken.sol` file:
+
+```bash
+cd src
+touch MyToken.sol
+```
+
+Open the file and add the following contract to it:
+
+```solidity
+--8<-- 'code/ethereum-api/foundry/erc20.sol'
+```
+
+Before you attempt to compile, install OpenZeppelin contracts as a dependency. You may have to commit previous changes to git beforehand. By default, Foundry uses git submodules instead of npm packages, so the traditional npm import path and command are not used. Instead, use the name of OpenZeppelin's Github repository:
```

Comment:
```suggestion
Before you attempt to compile, install OpenZeppelin contracts as a dependency. You may have to commit previous changes to git beforehand. By default, Foundry uses git submodules instead of npm packages, so the traditional npm import path and command are not used. Instead, use the name of OpenZeppelin's GitHub repository:
```

sorry! missed this one earlier